python/gravity_driven_flow.py

```diff
@@ -0,0 +1,39 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+import dolfin as dlfn
+
+from navier_stokes_problem import StationaryNavierStokesProblem, VelocityBCType
+
+from grid_generator import open_hyper_cube, HyperCubeBoundaryMarkers
+
+class GravityDrivenFlowProblem(StationaryNavierStokesProblem):
+    def __init__(self, n_points, main_dir = None):
+        super().__init__(main_dir)
+        
+        self._n_points = n_points
+        self._problem_name  = "OpenCube"
+        
+        self.set_parameters(1.0, 1.0)
+    
+    def setup_mesh(self):
+        # create mesh
+        openings = (("bottom", (0.5, 0.0), 0.5),
+                    ("top", (0.5, 1.0), 0.8))
+        self._mesh, self._boundary_markers = open_hyper_cube(2, self._n_points, openings)
+        self.write_boundary_markers()
+        
+    def set_boundary_conditions(self):
+        # velocity boundary conditions
+        velocity_bcs = (
+                (VelocityBCType.no_slip, HyperCubeBoundaryMarkers.left.value, None),
+                (VelocityBCType.no_slip, HyperCubeBoundaryMarkers.right.value, None),
+                (VelocityBCType.no_slip, HyperCubeBoundaryMarkers.bottom.value, None),
+                (VelocityBCType.no_slip, HyperCubeBoundaryMarkers.right.value, None))
+        self._bcs = {"velocity": velocity_bcs}
```

Comment:
I have three questions here.
1. Why only the velocity boundary condition is defined, but the pressure boundary condition is not defined here?
2. Why is the no-slip boundary condition defined twice on the right boundary?
3. What is the velocity boundary condition on the top opening area?

Reply:
1. If you want to add a pressure boundary condition you would need to define them and add them to the dictionary.
```python3
pressure_bcs = ( ... )
self._bcs = {"velocity": velocity_bcs, "pressure": pressure_bcs}
```
2. That is redundant and not necessary, my mistake. I fixed this in bfcc187.
3. There is no boundary condition, because I don't know any. You could, of course, apply an inflow velocity boundary at the top but then it is not more a channel flow rather then a gravity-driven flow. Here, the gravity accelerates the fluid.